llvm/lib/Target/AIE/AIELegalizerInfo.cpp

```diff
@@ -1188,6 +1181,103 @@ bool AIELegalizerInfo::legalizeG_FCMP(LegalizerHelper &Helper,
   return true;
 }
 
+unsigned getFCmpIntrID(CmpInst::Predicate Predicate, bool &SwapOperands) {
+  switch (Predicate) {
+  default:
+    llvm_unreachable("Unsupported FCmp predicate");
+  case CmpInst::FCMP_OGE:
+  case CmpInst::FCMP_UGE:
+    return Intrinsic::aie2_vgebf16;
+  case CmpInst::FCMP_OLT:
+  case CmpInst::FCMP_ULT:
+  case CmpInst::FCMP_UNE:
+  case CmpInst::FCMP_ONE:
+    return Intrinsic::aie2_vltbf16;
+  case CmpInst::FCMP_OGT:
+  case CmpInst::FCMP_UGT:
+    SwapOperands = true;
+    return Intrinsic::aie2_vltbf16;
+  case CmpInst::FCMP_OLE:
+  case CmpInst::FCMP_ULE:
+    SwapOperands = true;
+    return Intrinsic::aie2_vgebf16;
+  }
+}
+
+bool AIELegalizerInfo::legalizeG_FCMP(LegalizerHelper &Helper,
+                                      MachineInstr &MI) const {
+  MachineIRBuilder &MIRBuilder = Helper.MIRBuilder;
+  MachineRegisterInfo &MRI = *MIRBuilder.getMRI();
+  auto &CmpMI = cast<GFCmp>(MI);
+  const CmpInst::Predicate FPred = CmpMI.getCond();
+
+  const Register DstReg = CmpMI.getReg(0);
+  const Register LHS = CmpMI.getLHSReg();
+  const Register RHS = CmpMI.getRHSReg();
+  assert(MRI.getType(LHS) == MRI.getType(RHS) &&
+         "Mismatched operands for G_FCMP");
+
+  assert(CmpInst::isFPPredicate(FPred) && "Unsupported FCmp predicate");
+
+  if (FPred == CmpInst::FCMP_TRUE || FPred == CmpInst::FCMP_FALSE) {
+    MIRBuilder.buildConstant(DstReg, FPred == CmpInst::FCMP_TRUE ? 1 : 0);
+    MI.eraseFromParent();
+    return true;
+  }
+
+  unsigned LHSSize = MRI.getType(LHS).getSizeInBits();
+  if (LHSSize == 32)
+    return legalizeG_FCMP_FP32(Helper, MI, FPred);
+
+  assert(LHSSize == 16 && "Expected bf16 operands for FCmp");
+
+  const LLT V32S16 = LLT::fixed_vector(32, 16);
```

Comment:
Perhaps briefly describe how we're implementing it? In particular, that we insert both operands in a vector, which predicatess are directly supported, and how we transform the rest into these.